wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreatePasswordSecuredLink/CreateSecureGuestLinkViewController.swift

```diff
@@ -41,24 +55,79 @@ class CreateSecureGuestLinkViewController: UIViewController, CreatePasswordSecur
         return button
     }()
 
-    private lazy var viewModel: CreateSecureGuestLinkViewModel = {
-        CreateSecureGuestLinkViewModel(delegate: self)
+    private let setPasswordLabel: DynamicFontLabel = {
+        let label = DynamicFontLabel(
+            text: SecuredGuestLinkWithPasswordLocale.Textfield.header,
+            fontSpec: .subheadlineFont,
+            color: SemanticColors.Label.textFieldFloatingLabel
+        )
+        label.textAlignment = .left
+        return label
     }()
 
-    private let warningLabel: UILabel = {
-        var paragraphStyle = NSMutableParagraphStyle()
-        var label = UILabel()
+    private lazy var securedGuestLinkPasswordTextfield: ValidatedTextField = {
+        let textField = ValidatedTextField(
+            kind: .password(.guestLinkPassword, isNew: true),
+            leftInset: 8,
+            accessoryTrailingInset: 0,
+            cornerRadius: 12,
+            setNewColors: true,
+            style: .default
+        )
+        textField.addRevealButton(delegate: self)
+        textField.addTarget(self, action: #selector(handlePasswordValidation(for:)), for: .editingChanged)
```

Comment:
This is diverging from the intended use of `ValidatedTextField`. If you want to be informed of validation when the text changes, you should set a `TextFieldValidationDelegate`. You could register the view model as the delegate.

Reply:
You pushed the initial changes after our call and I followed up with a little bit of refactoring. Done!